src/game.rs

```diff
@@ -966,6 +966,117 @@ impl PostFlopGame {
         info.num_storage_actions += node.num_elements as u64;
     }
 
+    /// Reverse the action of push_chances
+    fn reverse_push_chances(&self, node: &PostFlopNode, info: &mut BuildTreeInfo) {
+        info.memory_usage_nodes += vec_memory_usage(&node.actions);
+        info.memory_usage_nodes += vec_memory_usage(&node.children);
+        info.num_storage_chances += (node.num_elements + node.num_elements_aux) as u64;
+    }
+
+    fn reverse_push_actions(&self, node: &PostFlopNode, info: &mut BuildTreeInfo) {
+        info.memory_usage_nodes += vec_memory_usage(&node.actions);
+        info.memory_usage_nodes += vec_memory_usage(&node.children);
+        info.num_storage_actions += node.num_elements as u64;
+    }
+
+    fn calculate_removed_line_info_recursive(&self, node: &PostFlopNode, info: &mut BuildTreeInfo) {
+        if node.is_terminal() {
+            return;
+        }
+
+        if node.is_chance() {
+            self.reverse_push_chances(node, info);
+        } else {
+            self.reverse_push_actions(node, info);
+        }
+    }
+
+    /// Remove lines after building the PostFlopGame but before allocating memory.
+    /// This allows the removal of chance-specific lines (e.g., remove overbets on
+    /// board-pairing turns) which we cannot do while building an action tree.
+    pub fn remove_lines(&mut self, lines: &Vec<Vec<Action>>) -> Result<(), String> {
```

Comment:
I prefer `&[Vec<Action>]` instead of `&Vec<Vec<Action>>`
```suggestion
    pub fn remove_lines(&mut self, lines: &[Vec<Action>]) -> Result<(), String> {
```

Reply:
:+1: